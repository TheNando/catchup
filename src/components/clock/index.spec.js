import { h } from 'preact'
import { Provider } from 'preact-redux'
import configureMockStore from 'redux-mock-store'
import { deep } from 'preact-render-spy'
import ConnectedClock, { Clock } from './index'

const pomodoro = {
  project: '1',
  duration: 1234,
  remaining: 1234,
}
const projects = ['1', '2']

const props = {
  pomodoro,
  projects,
  cachePomodoro: jest.fn(),
  completePomodoro: jest.fn(),
  setProject: jest.fn(),
}

const store = configureMockStore()({ dialog: {}, pomodoro, projects })

describe('Clock', () => {
  it('shows the time', () => {
    const context = deep(<Clock {...props} />)
    expect(context.find('#remaining').text()).toBe('20:34')
  })

  it('sets the selected project', () => {
    const context = deep(<Clock {...props} />)
    const mockEvent = { target: { selectedIndex: 2 } }
    expect(context.state().projectIndex).toEqual(1)
    context.find('select').simulate('change', mockEvent)
    expect(context.state().projectIndex).toEqual(2)
  })
})

describe('ConnectedClock', () => {
  it('receives projects from provider', () => {
    const context = deep(
      <Provider store={store}>
        <ConnectedClock />
      </Provider>
    )
    expect(context.output()).toMatchSnapshot()
  })
})
