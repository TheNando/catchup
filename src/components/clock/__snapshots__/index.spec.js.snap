// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedClock receives projects from provider 1`] = `
<div class="content">
  <div
    id="remaining"
    class="timer"
  >
    20:34
  </div>
  <select
    class="select"
    disabled={false}
    hintText="Project"
    outlined={true}
    box={false}
    selectedIndex={1}
    onChange={[Function anonymous]}
  >
    <option>1</option>
    <option>2</option>
  </select>
  <div class="actions">
    <button onClick={[Function anonymous]}>
      <i class="icon-button-large">play_circle_outline</i>
    </button>
  </div>
</div>
`;
